django/utils/html.py

```diff
@@ -17,7 +17,7 @@
 from .html_parser import HTMLParseError, HTMLParser
 
 # Configuration for urlize() function.
-TRAILING_PUNCTUATION = ['.', ',', ':', ';', '.)', '"', '\'', '!']
+TRAILING_PUNCTUATION_RE = re.compile(r'^(.*?)([.,:;!]+)$')
```

Comment:
Could you add a comment that explains the regular expression a bit? May not need to be as detailed as https://github.com/django/django/pull/6123 but that's an example of what I have in mind. It seems like some characters that were in `TRAILING_PUNCTUATION` aren't in the regex -- could you explain that to me so I understand it?